src/main/src/main/java/org/geoserver/ows/LanguageURLMangler.java

```diff
@@ -0,0 +1,37 @@
+/* (c) 2021 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+package org.geoserver.ows;
+
+import java.util.Map;
+
+/** Mangles service URL's based on the AcceptLanguages and Languages parameters */
+public class LanguageURLMangler implements URLMangler {
+
+    public static final String ACCEPT_LANGUAGES = "AcceptLanguages";
+    public static final String LANGUAGE = "Language";
+
+    @Override
+    public void mangleURL(
+            StringBuilder baseURL, StringBuilder path, Map<String, String> kvp, URLType type) {
+        if (Dispatcher.REQUEST.get() == null) return;
+
+        String language = "";
+        Map<String, String[]> parameterMap =
+                Dispatcher.REQUEST.get().getHttpRequest().getParameterMap();
+
+        for (Map.Entry<String, String[]> entry : parameterMap.entrySet()) {
```

Comment:
The String arrays of parameter is meant to be used if we expect repeated parameters in the requests eg.
AcceptLanguages=en&AcceptLanguages=it. What we should expect is instead AcceptLanguages=en,it or AcceptLanguaghes=en it.
I would suggest to retrieve before the  list of parameters name from the HttpsServletRequest and then do a request.getParameter(key). Then split the AcceptLanguages string value as needed

Reply:
Code first checks from parameterNames and matches AcceptLanguages from there.